### PR TITLE
Jacoco plugin report plugin

### PR DESCRIPTION
**Pull request by senthilhns**

### CODE REVIEW

File: plugin/jacoco_coverage_metrics.go [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+		MethodCoverageThreshold:      ParsePercentage(j.MethodCoverageThreshold),
+		ClassCoverageThreshold:       ParsePercentage(j.ClassCoverageThreshold),
+	}
+}
+
+func ParsePercentage(percentage string) float64 {
+	var value float64
+	_, err := fmt.Sscanf(percentage, "%f", &value)
+	if err != nil {
+		log.Fatalf("Error parsing percentage: %v", err)
```
Comment (Ompragash) [inline plugin/jacoco_coverage_metrics.go]: could you please replace the usage of `log` with `logrus` to adhere to the plugin dev guidelines?

**Review by Ompragash — COMMENTED** [review]
also, restructuring by moving tool specific logic into it's specific folder something like below:
```
/plugin
    /jacoco
        jacoco_plugin.go          //jacoco-specific logic for .exec files
        jacoco_coverage_metrics.go //jacoco metric calculations and threshold checks
    /jacoco_xml
        jacoco_xml_plugin.go      //jacoco-XML specific logic for .xml files
        jacoco_xml_metrics.go     //optional
    /cobertura
        cobertura_plugin.go       //Cobertura logic
        cobertura_metrics.go      //optional

keeping common functions like file handling, dir management, glob patterns in `util.go` and `plugin.go` containing shared structs and logic for setting thresholds, extracting reports, etc
```

File: docker/Dockerfile.linux.arm [hunk 2]
```diff
@@ -0,0 +1,10 @@
+FROM alpine:3.6 as alpine
```
Comment (Ompragash) [inline docker/Dockerfile.linux.arm]: we can delete this file as we only require `Dockerfile.linux.arm64`

File: docker/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,30 @@
+
```
Comment (Ompragash) [inline docker/Dockerfile]: please rename it to Dockerfile.linux.amd64

File: README.md [hunk 4]
```diff
@@ -1 +1,109 @@
-# drone-coverage-report
\ No newline at end of file
+A plugin to extract metrics from coverage reports generated by different coverage tools.
```
Comment (Ompragash) [inline README.md:1]: add a note in the README.md stating that this plugin can only be run on Linux (amd64/arm64) and explain why there won't be a Windows image for future reference

File: docker/manifest.tmpl [hunk 5]
```diff
@@ -0,0 +1,31 @@
+image: plugins/drone-coverage-report:{{#if build.tag}}{{trimPrefix "v" build.tag}}{{else}}latest{{/if}}
```
Comment (Ompragash) [inline docker/manifest.tmpl]: remove `drone-` suffix and update all image references to `plugins/coverage-report`

File: docker/manifest.tmpl [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    platform:
+      variant: v7
+      architecture: arm
+      os: linux
+  -
+    image: plugins/drone-coverage-report:{{#if build.tag}}{{trimPrefix "v" build.tag}}-{{/if}}linux-arm
+    platform:
+      variant: v6
+      architecture: arm
+      os: linux
```
Comment (Ompragash) [inline docker/manifest.tmpl]:
```suggestion
```
remove `linux-arm` and `linux-amd` references

**Review by Ompragash — CHANGES_REQUESTED** [review]

File: plugin/jacoco_coverage_metrics.go (hunk 1 above)
Comment (senthilhns) [reply plugin/jacoco_coverage_metrics.go]: fixed all log. usages to logrus.us

File: docker/Dockerfile.linux.arm (hunk 2 above)
Comment (senthilhns) [reply docker/Dockerfile.linux.arm]: done

File: docker/Dockerfile (hunk 3 above)
Comment (senthilhns) [reply docker/Dockerfile]: done

File: docker/manifest.tmpl (hunk 5 above)
Comment (senthilhns) [reply docker/manifest.tmpl]: done

File: README.md (hunk 4 above)
Comment (senthilhns) [reply README.md:1]: done

File: plugin/plugin_defs/defs.go [hunk 7]
```diff
@@ -0,0 +1,55 @@
+package plugin_defs
```
Comment (Ompragash) [inline plugin/plugin_defs/defs.go:1]: let’s keep this in `plugin.go`, including `pipeline.go` and `util.go` under `plugin/` dir, and only move the tool-specific logic to a separate folder

Comment (Ompragash) [reply plugin/plugin_defs/defs.go:1]: rest looks good

**Review by Ompragash — COMMENTED** [review]
<details>
  <summary>last check: tests are failing</summary>

```
go test ./...
go: downloading github.com/bmatcuk/doublestar/v4 v4.7.1
go: downloading golang.org/x/sys v0.0.0-20190422165155-953cdadca894
# github.com/harness-community/drone-coverage-report/plugin/jacoco
plugin/jacoco/jacoco_plugin.go:686:2: fmt.Println arg list ends with redundant newline
?   	github.com/harness-community/drone-coverage-report	[no test files]
FAIL	github.com/harness-community/drone-coverage-report/plugin/jacoco [build failed]
?   	github.com/harness-community/drone-coverage-report/plugin/plugin_defs	[no test files]
--- FAIL: TestSourcePathWithIncludeAndExclude (0.00s)
    plugin_test.go:119: Error in TestClassPathWithIncludeExclude: Error in GetBuildRootPath: Build
    plugin_test.go:98: Error in CheckFilesCopiedToWorkSpace: stat //sources/gameoflife-core/src/main/java/com/wakaleo/gameoflife/domain/Universe.java: no such file or directory
    plugin_test.go:98: Error in CheckFilesCopiedToWorkSpace: stat //sources/gameoflife-core/src/main/java/com/wakaleo/gameoflife/domain/Grid.java: no such file or directory
    plugin_test.go:98: Error in CheckFilesCopiedToWorkSpace: stat //sources/gameoflife-core/src/main/java/com/wakaleo/gameoflife/domain/Cell.java: no such file or directory
    plugin_test.go:98: Error in CheckFilesCopiedToWorkSpace: stat //sources/gameoflife-core/src/main/java/com/wakaleo/gameoflife/domain/GridReader.java: no such file or directory
    plugin_test.go:98: Error in CheckFilesCopiedToWorkSpace: stat //sources/gameoflife-core/src/main/java/com/wakaleo/gameoflife/domain/GridWriter.java: no such file or directory
    plugin_test.go:98: Error in CheckFilesCopiedToWorkSpace: stat //classes/pmd-rules.xml: no such file or directory
    plugin_test.go:98: Error in CheckFilesCopiedToWorkSpace: stat //classes/com/wakaleo/gameoflife/domain/Universe.class: no such file or directory
    plugin_test.go:98: Error in CheckFilesCopiedToWorkSpace: stat //classes/com/wakaleo/gameoflife/domain/Cell.class: no such file or directory
    plugin_test.go:98: Error in CheckFilesCopiedToWorkSpace: stat //classes/com/wakaleo/gameoflife/domain/GridReader.class: no such file or directory
    plugin_test.go:98: Error in CheckFilesCopiedToWorkSpace: stat //classes/com/wakaleo/gameoflife/domain/GridWriter.class: no such file or directory
    plugin_test.go:98: Error in CheckFilesCopiedToWorkSpace: stat //classes/com/wakaleo/gameoflife/domain/Grid.class: no such file or directory
    plugin_test.go:98: Error in CheckFilesCopiedToWorkSpace: stat //classes/custom-checkstyle.xml: no such file or directory
    plugin_test.go:98: Error in CheckFilesCopiedToWorkSpace: stat //execFiles/gameoflife-core/target/jacoco.exec: no such file or directory
    plugin_test.go:98: Error in CheckFilesCopiedToWorkSpace: stat //execFiles/gameoflife-web/target/jacoco.exec: no such file or directory
FAIL
FAIL	github.com/harness-community/drone-coverage-report/plugin	1.693s
FAIL
```

</details>

File: scripts/build.sh [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+
+# disable cgo
+export CGO_ENABLED=0
+
+set -e
+set -x
+
+# linux
+GOOS=linux GOARCH=amd64 go build -o release/linux/amd64/plugin
+#GOOS=linux GOARCH=arm64 go build -o release/linux/arm64/plugin
```
Comment (Ompragash) [inline scripts/build.sh]:
```suggestion
GOOS=linux GOARCH=arm64 go build -o release/linux/arm64/plugin
```

**Review by Ompragash — CHANGES_REQUESTED** [review]

**Review by Ompragash — APPROVED** [review]

**Pull request merged**